mods/mc_student/init.lua

```diff
@@ -265,38 +348,43 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 	end
 
 	if formname == "mc_student:marker" then
-		if fields.back then
-			show_student_menu(player)
-		elseif fields.message then
+		if fields.message then
 			place_marker(player,fields.message)
 		elseif fields.message == nil then
 			return true
 		end
 	end
 
 	if formname == "mc_student:coordinates" then
-		if fields.back then
-			show_student_menu(player)
-		elseif fields.record then
-			record_coordinates(player,fields.message)
+		local event = minetest.explode_textlist_event(fields.coordlist)
+
+		if fields.record then
+			record_coordinates(player,fields.note)
 		elseif fields.coordlist then
-			local event = minetest.explode_textlist_event(fields.coordlist)
-		    	if event.type == "CHG" then
-			-- "CHG" = something is selected in the list
-				context.selected = event.index
-		    	end
+			if event.type == "CHG" then
+				selectedCoord = event.index
+			end
 		elseif fields.go then
-			local pname = player:get_player_name()
-			local pmeta = player:get_meta()
-			if not context.selected then
-				context.selected = 1
+			if selectedCoord ~= 0 then
+				local new_pos = minetest.deserialize(pmeta:get_string("coordinates")).coords[selectedCoord]
+				player:set_pos(new_pos)
 			end
-		    	local temp = minetest.deserialize(pmeta:get_string("coordinates"))
-		    	local new_pos_char = temp.coords[context.selected]
-			local new_pos_tab = pos_split(new_pos_char)
-		    	player:set_pos(new_pos_tab)
-		elseif fields.deleteall then
-			local pmeta = player:get_meta()
+		elseif fields.delete then
+			local data = minetest.deserialize(pmeta:get_string("coordinates"))
+			local newCoords, newNotes, newRealms = {}, {}, {}	
+
+			for i,coord in ipairs(data.coords) do
```

Comment:
This can crash the game when the coordlist is empty and the player clicks the `Delete` button.
Also, the when there is only one item (coordinate) in the list, the `Delete` button will not remove it and only the `Clear All` button removes the coordinate.

`AsyncErr: Lua: Runtime error from mod 'mc_student' in callback on_playerReceiveFields(): ...assroom-update-student-notebook\mods\mc_student\init.lua:376: attempt to index local 'data' (a nil value)`